src/main/java/pl/touk/sputnik/processor/shellcheck/ShellcheckResultParser.java

```diff
@@ -0,0 +1,64 @@
+package pl.touk.sputnik.processor.shellcheck;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import pl.touk.sputnik.processor.shellcheck.json.ShellcheckMessage;
+import pl.touk.sputnik.processor.tools.externalprocess.ExternalProcessResultParser;
+import pl.touk.sputnik.review.Severity;
+import pl.touk.sputnik.review.Violation;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+class ShellcheckResultParser implements ExternalProcessResultParser {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final String SHELLCHECK_ERROR = "error";
+    private static final String SHELLCHECK_WARNING = "warning";
+    private static final String SHELLCHECK_INFO = "info";
+
+    @Override
+    public List<Violation> parse(String shellcheckOutput) {
+        if (StringUtils.isEmpty(shellcheckOutput)) {
+            return Collections.emptyList();
+        }
+        try {
+            List<Violation> violations = new ArrayList<>();
```

Comment:
Would it be more readable with Java Stream API?

Reply:
Yes, done